dsn.go

```diff
@@ -515,6 +523,38 @@ func fillMissingConfigParameters(cfg *Config) error {
 	return nil
 }
 
+func extractDomainFromHost(host string) (domain string, index int) {
+	i := strings.LastIndex(strings.ToLower(host), topLevelDomainPrefix)
+	if i >= 1 {
+		domain = host[i:]
+		return domain, i
+	}
+	return "", i
+}
+
+func getDomainBasedOnRegion(region string) string {
+	if strings.HasPrefix(strings.ToLower(region), "cn-") {
+		return cnDomain
+	}
+	return defaultDomain
+}
+
+func extractRegionFromAccount(account string) (region string, posDot int) {
```

Comment:
Is it compliant with https://docs.snowflake.com/en/user-guide/admin-account-identifier#format-1-preferred-account-name-in-your-organization ?

Reply:
I think it is, since the `.` org separator is used `for SQL commands and operations`. Also there are explicit tests that confirm that in dsn_test.go